allow option to define relationship type in /goterm/<id>/genes/ endpoint

deepakunni3 opened the pull request:
Fix for #138.

**Note:** This is a temporary fix until https://github.com/geneontology/amigo/pull/469 and https://github.com/owlcollab/owltools/issues/241 are implemented.

deepakunni3 committed:
allow option to define relationship type in /goterm/<id>/genes/ endpoint

cmungall reviewed (approved):
The canonical RO names that we should expose in the API on the left. The amigo golr field name on the right

`involved_in` => `isa_partof_closure`
`involved_in_regulation_of` => `regulates_closure - isa_partof_closure`
`acts_upstream_of_or_within` => `regulates_closure`

deepakunni3 committed:
use proper relationship_type names

involved_in => isa_partof_closure
involved_in_regulation_of => regulates_closure - isa_partof_closure
acts_upstream_of_or_within => regulates_closure

deepakunni3 commented:
@cmungall Added suggested changes
